noir/compiler/noirc_evaluator/src/ssa/acir_gen/acir_ir/big_int.rs

```diff
@@ -0,0 +1,54 @@
+use acvm::FieldElement;
+use num_bigint::BigUint;
+
+/// Represents a bigint value in the form (id, modulus) where
+///     id is the identifier of the big integer number, and
+///     modulus is the identifier of the big integer size
+#[derive(Default, Clone, Copy, Debug)]
+pub(crate) struct BigIntId(pub(crate) u32, pub(crate) u32); //bigint id, modulus id
+
+impl BigIntId {
+    pub(crate) fn as_field(&self) -> FieldElement {
+        FieldElement::from(self.0 as u128)
+    }
+
+    pub(crate) fn modulus_id(&self) -> FieldElement {
+        FieldElement::from(self.1 as u128)
+    }
+}
+
+/// BigIntContext is used to generate identifiers for big integers and their modulus
+#[derive(Default, Debug)]
+pub(crate) struct BigIntContext {
+    modulus: Vec<BigUint>,
+    big_integers: Vec<BigIntId>,
+}
+
+impl BigIntContext {
+    /// Creates a new BigIntId for the given modulus and returns it.
+    pub(crate) fn new_big_int(&mut self, modulus: FieldElement) -> BigIntId {
+        let id = self.big_integers.len() as u32;
+        let result = BigIntId(id, modulus.to_u128() as u32);
```

Comment:
Doesn't this restrict the modulus to be a u128 for a bigint?

Reply:
If its the modulus_id, then I think we should rename `modulus` to `modulus_id` and probably use `u32`

Reply:
Spoke with Guillaume and this is mainly a naming issue

Reply:
It is the modulus_id, I changed to modulus_id